lm_engine/mixins/dense/base.py

```
@@ -137,186 +124,86 @@ def forward(
         self,
         input_ids: torch.Tensor | None = None,
         cache_params: GenerationCache | None = None,
-        attention_mask: torch.Tensor | None = None,
-        position_ids: torch.Tensor | None = None,
-        use_cache: bool | None = None,
-        cu_seqlens: torch.Tensor | None = None,
-        max_seqlen: int | None = None,
+        attention_mask_info: AttentionMaskInfo = AttentionMaskInfo(),
```

Ruff (0.16.0)

[warning] Do not perform function call `AttentionMaskInfo` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)